src/polyfem/mesh/Mesh.hpp

```diff
@@ -440,6 +440,12 @@ namespace polyfem
 				else
 					return 0;
 			}
+			/// @brief Get the volume selection of all elements (cells in 3d, faces in 2d)
+			/// @return Const reference to the vector of body IDs
+			virtual const std::vector<int> &get_body_ids() const
```

Comment:
why virtual?

Reply:
Only because `virtual int get_body_id(const int primitive) const` is `virtual`. We can drop `virtual` if you want.